src/crawlee/http_crawler/_http_crawler.py

```diff
@@ -1,77 +1,61 @@
 from __future__ import annotations
 
 import logging
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Generic
 
-from crawlee._utils.docs import docs_group
+from pydantic import ValidationError
+
+from crawlee import EnqueueStrategy
+from crawlee._request import BaseRequestData
+from crawlee._utils.urls import convert_to_absolute_url, is_url_absolute
 from crawlee.basic_crawler import BasicCrawler, BasicCrawlerOptions, ContextPipeline
 from crawlee.errors import SessionError
 from crawlee.http_clients import HttpxHttpClient
-from crawlee.http_crawler._http_crawling_context import HttpCrawlingContext
+
+from ._http_crawling_context import HttpCrawlingContext, ParsedHttpCrawlingContext, TParseResult
+from ._http_parser import NoParser, StaticContentParser
 
 if TYPE_CHECKING:
     from collections.abc import AsyncGenerator, Iterable
 
-    from typing_extensions import Unpack
+    from typing_extensions import Any, Unpack
+
+    from crawlee._types import BasicCrawlingContext, EnqueueLinksFunction, EnqueueLinksKwargs
 
-    from crawlee._types import BasicCrawlingContext
+logger = logging.getLogger(__name__)
 
 
-@docs_group('Classes')
-class HttpCrawler(BasicCrawler[HttpCrawlingContext]):
+class HttpCrawlerGeneric(Generic[TParseResult], BasicCrawler[ParsedHttpCrawlingContext[TParseResult]]):
```

Comment:
This one I tried. Unfortunately I do not think it is possible (and stay statically type consistent) due to the TParseResult deciding about the type of TCrawlingContext ( TCrawlingContext == ParsedHttpCrawlingContext[TParseResult] ). In other words, TCrawlingContext is directly dependent on TParseResult and as such they can't be specified as independent generics parameters.  The fact that they would match correctly in only some cases will be treated by mypy as error.

Further "generic" extensions of BeautifulSoupCrawler will not be possible. But since all functionality (except for the parser) is already in HttpCrawlerGeneric then I do not see any use for extending BeautifulSoupCrawler.

Reply:
Done